packages/amplify-category-auth/resources/cloudformation-templates/auth-template.yml.ejs

```diff
@@ -1,4 +1,5 @@
 <% var autoVerifiedAttributes = props.autoVerifiedAttributes ? props.autoVerifiedAttributes.concat(props.aliasAttributes).filter((attr, i, aliasAttributeArray) => ['email', 'phone_number'].includes(attr) && aliasAttributeArray.indexOf(attr) === i) : [] %>
+<% var configureSMS = ((props.autoVerifiedAttributes && props.autoVerifiedAttributes.includes('phone_number')) || (props.mfaConfiguration != 'OFF' && props.mfaTypes && props.mfaTypes.includes('SMS Text Message')) || (props.requiredAttributes && props.requiredAttributes.includes('phone_number'))) %> 
```

Comment:
This seems to be a complex check. could you please provide this as a prop.

Reply:
any time complex logic starts developing in an .ejs file, I wonder if it has to be in here, or if it can be moved up a layer. I looked into it a little bit and it felt it like based on the way layers of above are extracted it feels easier to do it in here, even though it's not ideal.

Reply:
I think there is an issue with providing it as a prop

Reply:
yeah I agree, looks like we can pass it in now. We could resolve this when moving away from ejs.